app/models/manageiq/providers/inventory/persister/builder.rb

```diff
@@ -86,6 +84,7 @@ def construct_data
         add_properties(@shared_properties, :if_missing)
 
         send(@name.to_sym) if @name.respond_to?(:to_sym) && respond_to?(@name.to_sym)
+        yield(self) if block_given?
```

Comment:
is construct_data ever called without a block after these changes? https://github.com/ManageIQ/manageiq/pull/21597/files#diff-df074682b12136ada06775bc98aa0ff8a4d47cc470d2845ad3ba603952f2c334R40

(can't tell if construct_data is public or private here)

Reply:
Oh I see..if no block is given to `prepare_data`, then it will forward a nil block...I'm not sure I've ever seen that before :)

Reply:
FYI I had to move where the block was called to before we called auto_model_ckass because there were some blocks that set model class but it had already raised an exception